### ISSUE TITLE
[compat API] unable to prune images with a dangling filter

### ISSUE THREAD

**Issue by benoitf** [containers/podman#17614]
### Issue Description

I'm unable to use the compat API (docker API) with Podman for the pruneImage command
https://docs.docker.com/engine/api/v1.42/#tag/Image/operation/ImagePrune

### Steps to reproduce the issue

Steps to reproduce the issue
1. pull an image (assuming it's not already used/there before)
2. ```
   podman pull quay.io/podman/hello
   ```
3. call the REST API


```shell
curl -X POST --unix-socket /var/run/docker.sock "http:/v1.41/images/prune?filters=\{\"dangling\":\{\"false\":true\}\}" 
```

### Describe the results you received

I got
```
{"cause":"specifying \"dangling\" filter more than once with different values is not supported","message":"specifying \"dangling\" filter more than once with different values is not supported","response":500}
```

### Describe the results you expected

I should receive something like:

```
{"ImagesDeleted":[{"Untagged":"quay.io/podman/hello:latest"},....
```

### podman info output

```yaml
Client:       Podman Engine
Version:      4.4.1
API Version:  4.4.1
Go Version:   go1.19
Git Commit:   34e8f3933242f2e566bbbbf343cf69b7d506c1cf
Built:        Wed Feb  8 23:21:26 2023
OS/Arch:      darwin/amd64

Server:       Podman Engine
Version:      4.4.1
API Version:  4.4.1
Go Version:   go1.19.5
Built:        Thu Feb  9 11:58:53 2023
OS/Arch:      linux/amd64
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by Luap99** [comment]
@vrothberg PTAL

Problems seems to be that podman adds the dangling filter here: https://github.com/containers/podman/blob/a54a4041c5d5094c5925f2be4ef09a383c16f53b/pkg/domain/infra/abi/images.go#L58
and thus validation in libimage complains about it.

This can also be reproduced with `podman image prune -f --filter dangling=false`

**Comment by benoitf** [comment]
@vrothberg is it possible to grant collaborator access to @vyasgun ?

**Comment by cristianrgreco** [comment]
Seems this isn't fixed as of the latest 4.5.0 - in what version has this been released?
